src/client.js

```diff
@@ -131,6 +149,20 @@ let pushClient = {
     navigator.serviceWorker.removeEventListener('message', messageHandler);
   },
 
+  async getSubscription() {
+    if (!this.supported()) {
+      return;
```

Comment:
I'm thinking if the developer is calling these functions without checking for support, then something very wrong is happening - would throwing an error be to much here?